platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1267,49 +1076,7 @@ public void removeAllMarkers() {
      * @param listener The {@link MapChangeListener} to call when the map change events occur due to camera updates or user interaction
      */
     public void setMapChangeListener(@Nullable final MapChangeListener listener) {
-        final Runnable regionIsChanging = (listener == null) ? null : new Runnable() {
-            @Override
-            public void run() {
-                listener.onRegionIsChanging();
-            }
-        };
-
-        mapChangeListener = (listener == null) ? null : new MapChangeListener() {
-            @Override
-            public void onViewComplete() {
-                uiThreadHandler.post(new Runnable() {
-                    @Override
-                    public void run() {
-                        listener.onViewComplete();
-                    }
-                });
-            }
-
-            @Override
-            public void onRegionWillChange(final boolean animated) {
-                uiThreadHandler.post(new Runnable() {
-                    @Override
-                    public void run() {
-                        listener.onRegionWillChange(animated);
-                    }
-                });
-            }
-
-            @Override
-            public void onRegionIsChanging() {
-                uiThreadHandler.post(regionIsChanging);
-            }
-
-            @Override
-            public void onRegionDidChange(final boolean animated) {
-                uiThreadHandler.post(new Runnable() {
-                    @Override
-                    public void run() {
-                        listener.onRegionDidChange(animated);
-                    }
-                });
-            }
-        };
+        mapChangeListener = listener;
```

Comment:
Shouldn't tangram library not make sure there are posted on the ui thread, or make it flexible for client's to aptly handle this.

Open to either but make sense to properly document this.

Reply:
Currently we invoke the mapChangeListener from two places:
- gesture listeners, which are always invoked on the UI thread
- camera updates, which the client application invokes
Handling client invocations of camera updates off the UI thread would be tricky. If we post the event to the UI thread and then immediately execute the camera update then we can't be sure that the event was executed before the camera changed. It seems like we're stuck with a choice between making sure the event is on the UI thread and making sure the event happens before the camera change.